### PR TITLE
Doc link issue workaround

### PR DESCRIPTION
**Pull request by sloriot**
Fixes project name linking to CGAL namespace reported in #5600

### CODE REVIEW

File: Documentation/doc/Documentation/Doxyfile.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # this package is a little peculiar and different from the other packages
 @INCLUDE = ${CGAL_DOC_PACKAGE_DEFAULTS}
-PROJECT_NAME = "CGAL ${CGAL_DOC_VERSION} - Manual"
+PROJECT_NAME = "C&zwj;GAL ${CGAL_DOC_VERSION} - Manual"
```
Comment (lrineau) [inline Documentation/doc/Documentation/Doxyfile.in:3]: I am against that patch. The [zero-width joiner](https://en.wikipedia.org/wiki/Zero-width_joiner) is not made for that sort of thing. "C ZWJ G" should display one letter that is the union of C and G in one letter.

Maybe the [zero-width non-joiner](https://en.wikipedia.org/wiki/Zero-width_non-joiner) could be used, but that is a very nasty visual trick: "C ZWNJ G" should display a ligature of C and G. There might be fonts were that ligature exist.

**Review by lrineau — CHANGES_REQUESTED** [review]

File: Documentation/doc/Documentation/Doxyfile.in (hunk 1 above)
Comment (lrineau) [reply Documentation/doc/Documentation/Doxyfile.in:3]: Even with the [zero-width space](https://en.wikipedia.org/wiki/Zero-width_space) the effect will be to break CGAL into two words: "C" and "GAL", but joined by a zero-width space. Even that is a nasty visual trick. Why this workaround, by the way?

**Comment by sloriot** [comment]
Do you prefer going with a %CGAL removed with post-processing?

**Comment by lrineau** [comment]
> Do you prefer going with a %CGAL removed with post-processing?

I think that would be a better option than using special spaces incorrectly.

**Comment by sloriot** [comment]
Fixed by the latest commit. However I notice that the navtree of the main page is re-broken
Screenshot from 2021-06-29 18-41-46
There is an extra duplicated layer. Maybe it has to do with our navtree.js hacks...

**Comment by sloriot** [comment]
Does not seem related. Commenting our hacks does not change anything. @albert-github do you know if it can be a regression in doxygen?

**Comment by albert-github** [comment]
I didn't yet dive into it but from which version is the picture shown?

**Comment by albert-github** [comment]
I dove a little bit in it. The choice for `&zwj;` or maybe better `&zwnj;` is because they are handled properly in the headers and index lists whilst this is not the case with the `%` sign (also commands are not handled in the headers / index lists). The reason for using the `%CGAL` is to make a distinction between the CGAL product and the CGAL namespace (without the `%` there might be an unintended link to the namespace).

Although `&zwj;` / `&zwnj;` are not that nice I personally favor them over postprocessing steps as shown in this patch.
(This is not an easy problem to fix).

**Comment by sloriot** [comment]
> I didn't yet dive into it but from which version is the picture shown? and what is the page name `<pck>/<file>.html`?

I'm using Doxygen master latest HEAD. Actually, I tried it again and I noticed that it is the fact of adding the % that is breaking the navtree. 

I don't know what to do. @lrineau ?

**Comment by lrineau** [comment]
The incorrect usage of "invisible spaces" is really something I do not want to see spread. Would it work to have a macro `\cgal`, to design CGAL as a library or a project, and then `CGAL` would be restricted to the CGAL namespace? That macro `\cgal` could use a [zero-width non-joiner](https://en.wikipedia.org/wiki/Zero-width_non-joiner). The issue with that zero-width non-joiner, is that the browser will be allowed to add a line break between "C" and "GAL".

**Comment by sloriot** [comment]
The macro won't work for the project name as it is used in doxygen config file. Let me try to disable the link to CGAL.

**Comment by sloriot** [comment]
I was thinking of using `EXCLUDE_SYMBOLS` but then the namespace CGAL is totally ignored so it is not a good solution. We could disable automatic link with  `AUTOLINK_SUPPORT=OFF` but then we would require to explicitly add `\ref` everywhere we want a link to be generated. Still no solution but the one suggested by @albert-github with drawbacks mentioned by @lrineau.

**Comment by lrineau** [comment]
@sloriot This is the third (and last PR) I have revived. It was closed by error when I deleted the branch `5.1.x-branch` from Github. I have reassigned it to `5.2.x-branch`.

**Pull request closed without merging**